File: BinarySearch/153.Find Minimum in Rotated Sorted Array/Solution.py
class Solution:
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0:
            return None

        left = 0
        right = len(nums) - 1

        while left < right - 1:
            mid = left + (right - left) // 2
            if nums[mid] > nums[len(nums)-1]:
                left = mid
            else:
                right = mid

        return nums[left] if nums[left] <= nums[right] else nums[right]

File: BinarySearch/153.Find Minimum in Rotated Sorted Array/Solution.java
class Solution {
    public int findMin(int[] nums) {
        if(nums.length == 0) return -1;
        
        int left = 0;
        int right = nums.length - 1;
        while(left < right -1){
            int mid = left + (right - left)/2;
            if(nums[mid]>nums[nums.length-1]){
                left = mid;
            }else {
                right = mid;
            }
        }
        
        return Math.min(nums[left], nums[right]);
    }
}
